libsignal-service/src/provisioning/mod.rs

```diff
@@ -211,18 +213,48 @@ pub async fn link_device<
             },
         )?;
 
-        let aci_key_pair = KeyPair::new(aci_public_key, aci_private_key);
-        let pni_key_pair = KeyPair::new(pni_public_key, pni_private_key);
-
+        let aci_key_pair =
+            IdentityKeyPair::new(aci_public_key, aci_private_key);
+        let pni_key_pair =
+            IdentityKeyPair::new(pni_public_key, pni_private_key);
+
+        let (
+            _aci_pre_keys,
+            aci_signed_pre_key,
+            _aci_pq_pre_keys,
+            aci_pq_last_resort_pre_key,
+        ) = crate::pre_keys::generate_pre_keys(
+            aci_store,
+            &aci_key_pair,
+            csprng,
+            true,
+            0,
+            0,
+        )
+        .await?;
         let aci_pq_last_resort_pre_key =
-            generate_last_resort_kyber_key(aci_store, &aci_key_pair).await?;
+            aci_pq_last_resort_pre_key.expect("requested last resort key");
+        assert!(_aci_pre_keys.is_empty());
+        assert!(_aci_pq_pre_keys.is_empty());
+
+        let (
+            _pni_pre_keys,
+            pni_signed_pre_key,
+            _pni_pq_pre_keys,
+            pni_pq_last_resort_pre_key,
+        ) = crate::pre_keys::generate_pre_keys(
```

Comment:
This might be incorrect: we're using this to generate pre keys for other devices, so we *don't* have to store them ourselves, no? Needs a double check.

Reply:
Double check: we're generating them for ourselves, so this is fine!